sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/GpuBroadcastExchangeExec.scala

```diff
@@ -111,30 +113,37 @@ class SerializeConcatHostBuffersDeserializeBatch(
       }
       assert(headers.length <= 1 && buffers.length <= 1)
       withResource(new NvtxRange("broadcast manifest batch", NvtxColor.PURPLE)) { _ =>
-        if (headers.isEmpty) {
-          batchInternal = GpuColumnVector.emptyBatchFromTypes(dataTypes)
-          GpuColumnVector.extractBases(batchInternal).foreach(_.noWarnLeakExpected())
-        } else {
-          withResource(JCudfSerialization.readTableFrom(headers.head, buffers.head)) { tableInfo =>
-            val table = tableInfo.getContiguousTable
-            if (table == null) {
-              val numRows = tableInfo.getNumRows
-              batchInternal = new ColumnarBatch(new Array[ColumnVector](0), numRows)
-            } else {
-              batchInternal = GpuColumnVectorFromBuffer.from(table, dataTypes)
-              GpuColumnVector.extractBases(batchInternal).foreach(_.noWarnLeakExpected())
-              table.getBuffer.noWarnLeakExpected()
+        try {
+          val res = if (headers.isEmpty) {
+            val gpuCV = GpuColumnVector.emptyBatchFromTypes(dataTypes)
+            GpuColumnVector.extractBases(gpuCV).foreach(_.noWarnLeakExpected())
```

Comment:
nit: SpillableColumnarBatch can handle this use case, and also the empty ColumnarBatch use case below.  I think we could make this API and the code a little bit cleaner.  But if you want to keep it as is, just because of time constraints I am fine with it.